kubetest/extract_k8s.go

```diff
@@ -373,6 +373,23 @@ func (e extractStrategy) Extract(project, zone string, extractSrc bool) error {
 		if zone == "" {
 			return fmt.Errorf("--gcp-zone unset")
 		}
+		if e.option == "gke" {
+			log.Print("*** --extract=gke is deprecated, migrate to --extract=gke-default ***")
+		}
+		if strings.HasSuffix(e.option, "-latest") {
+			// get latest supported master version
+			res, err := output(exec.Command("gcloud", "container", "get-server-config", fmt.Sprintf("--project=%v", project), fmt.Sprintf("--zone=%v", zone), "--format=value(validMasterVersions)"))
+			if err != nil {
+				return err
+			}
+			versions := strings.Split(string(res), ";")
+			if len(versions) == 0 {
+				return fmt.Errorf("invalid gke master version string: %s", string(res))
+			}
+			return setReleaseFromGcs("kubernetes-release/release", versions[0], extractSrc)
+		}
+
+		// prod & staging & test
 		ci, err := output(exec.Command("gcloud", "container", "get-server-config", fmt.Sprintf("--project=%v", project), fmt.Sprintf("--zone=%v", zone), "--format=value(defaultClusterVersion)"))
```

Comment:
Do we need to do this in case of "-latest"?

Reply:
when validMasterVersions is missing?

Reply:
Sorry, didn't see all the `return`s in the `if` block.